src/framework/server.ts

```diff
@@ -76,8 +61,46 @@ export type SettingsInput = createExpressGraphql.OptionsData &
     plugins: Plugin.RuntimeContributions[]
     contextContributors: ContextContributor<any>[]
   }
+/**
+ * [API Reference](https://nexus-future.now.sh/#/references/api?id=server)  ⌁  [Guide](todo)
+ *
+ * ### todo
+ *
+ */
+export interface Server {
+  /**
+   * todo
+   */
+  start: () => Promise<void>
+  /**
+   * todo
+   */
+  stop: () => Promise<void>
+}
+
+interface ServerWithInstance<T extends any> extends Server {
+  instance: T
+}
+
+interface CustomServerHookInput {
```

Comment:
This feedback invalidates some of the below. Actually a bit tricky of an API/naming here, so pardon the multiple takes I'm giving...

- I don't think we need to pass `settings` reference because user can get it from lexical scope.

- I think it might be simplest to just do:

	```ts
	server.custom(({ express }) => {
	  
	}
	```



<details>
<summary>something else, not sure...</summary>

```ts
// simple case
server.custom(express => {
  express.use(...)
  return {
    express.listen(settings.current.server.port)
  }
})
```
```ts
// advanced case
server.custom(express => {
  express.use(...)
  return {
    start(){
      // user can lexically access settings, should be fine!
      express.listen(settings.current.server.port)
    }
  }
})
```
```ts
// most advanced case
server.custom((_, { schema }) => {
  // User can get at schema, may ignore originalServer instance if they wish
  // I don't think we actually need the `settings` here, should be 
  // good enough with lexical scope access.
})
```

or...

```ts
// most advanced case
server.replace(({ schema }) => {
  // User can get at schema, may ignore originalServer instance if they wish
  // I don't think we actually need the `settings` here, should be 
  // good enough with lexical scope access.
})
```


- It supports (by way of simplest api/syntax) a primary use-case of configuring express, not swapping the server completely.
- It allows user to ***name*** the server as it actually is
- `.replace` can improve boot performance because we can skip creating `express` instance completely...
  - Then we need to make it a runtime error if user calls both `replace` and `custom`...
  - Basically back to raw 🙈 
</details>

Reply:
Actually, I'm reconsidering your suggestion. I'm fine with calling it `express`, but I think I'd prefer keeping express of type `Server` and not the raw `Express`. It makes the usage way simpler in case you just want to add middlewares and other kind of stuff to the express instance. All you need to do is mutate the `express` (previously `originalServer`) property, and return it.

As to your suggestion for `replace/custom`, I personally find it easier and more generic the way it is now. There's no need to wonder what's the difference between `replace` and `custom`, no possibility of errors if you actually use both, and, with my suggestion just above, it makes it very simple to just configure express

```js
server.custom(({ originalServer }) => {
  originalServer.instance.use(/*some custom middleware*/)

  return originalServer
})
```

Reply:
How about we make it so returning is optional and if you don’t it means do not override the server? I think this is most clear. It does create more room for user error though. If we had replace and custom then we could make more precise return type. But not sure it’s worth the extra dpi surface. Think the return-to-replace is the sweet spot.

Reply:
> I'd prefer keeping express of type Server and not the raw Express. It makes the usage way simpler in case you just want to add middlewares and other kind of stuff to the express instance. 

Then user should never need access to server interface. Simplest of all?